pkg/gui/controllers/commits_files_controller.go

```diff
@@ -176,43 +178,102 @@ func (self *CommitFilesController) checkout(node *filetree.CommitFileNode) error
 	return self.c.Refresh(types.RefreshOptions{Mode: types.ASYNC})
 }
 
-func (self *CommitFilesController) discard(node *filetree.CommitFileNode) error {
+func (self *CommitFilesController) discard(selectedNodes []*filetree.CommitFileNode) error {
 	parentContext, ok := self.c.CurrentContext().GetParentContext()
 	if !ok || parentContext.GetKey() != context.LOCAL_COMMITS_CONTEXT_KEY {
 		return self.c.ErrorMsg(self.c.Tr.CanOnlyDiscardFromLocalCommits)
 	}
 
-	if node.File == nil {
-		return self.c.ErrorMsg(self.c.Tr.DiscardNotSupportedForDirectory)
-	}
+	removeFileRange := func() error {
+		return self.c.WithWaitingStatus(self.c.Tr.UpdatingPatch, func(gocui.Task) error {
+			if !self.c.Git().Patch.PatchBuilder.Active() {
+				if err := self.startPatchBuilder(); err != nil {
```

Comment:
I think that for this we should actually use the original logic in `DiscardOldFileChanges` but with multiple files. With that logic (which does not depend on patches) we can support additions/modifications/deletions of binary files.

Given that, we don't to reset the current patch, though we should anyway given that the user won't be able to continue adding to it because the original commit will be gone. But for that, rather than have a second prompt, I would just add an addendum to the first prompt saying 'This will reset the current custom patch'

Reply:
Oh wow good call! I didn't even think about binary files. I'll take care of that & add the custom patch reset note to the prompt.